assignments/this.js

```diff
@@ -1,26 +1,81 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. Implicit Binding - When a funtion in invoced, if you look to the left of the dot, that is what is being referenced.
+* 2. Explicit Binding - This occurs when .call()- pass one by one, .apply()-pass on with an array, or .bind()- same as .call but returns new function to invoke later,  are used in a function
+* 3. New Binding - when a function is invoked with the new keyword, the this keyword inside that function is bound to the new object being constructed.
+* 4. Window Binding - if non of the rules apply, the this key word will default to the windo object.
```

Comment:
I like your answers to the 'this' questions! Also, you did well on the explanation examples below.